code/game/machinery/big_manipulator/big_manipulator.dm

```diff
@@ -0,0 +1,909 @@
+	balloon_alert(user, "successfully [id_lock ? "locked" : "unlocked"]")
+
+/// Attaching the arm effect to the core.
+/obj/machinery/big_manipulator/proc/create_manipulator_arm()
+	manipulator_arm = new/obj/effect/big_manipulator_arm(src)
+	manipulator_arm.dir = NORTH
+	vis_contents += manipulator_arm
+
+/// Destroying the manipulator if the arm is destroyed.
+/obj/machinery/big_manipulator/proc/on_hand_qdel()
+	SIGNAL_HANDLER
+
+	deconstruct(TRUE)
+
+/obj/machinery/big_manipulator/proc/toggle_power_state(mob/user)
+	var/new_power_state = !on
+
+	if(!user)
+		on = new_power_state
+		if(!on)
+			remove_all_huds()
+		return
+
+	if(new_power_state)
+		if(!powered())
+			balloon_alert(user, "no power!")
+			return
+
+		if(!anchored)
+			balloon_alert(user, "anchor first!")
+			return
+
+		validate_all_points()
+
+		on = new_power_state
+		SStgui.update_uis(src)
+		try_kickstart(user)
+
+	else
+		drop_held_atom()
+		on = new_power_state
+		cycle_timer_running = FALSE
+		// Set stopping task instead of ending current task immediately
+		if(current_task != CURRENT_TASK_NONE && current_task != CURRENT_TASK_STOPPING)
+			start_task(CURRENT_TASK_STOPPING, 0)
+			// Schedule automatic completion of stopping task
+			addtimer(CALLBACK(src, PROC_REF(complete_stopping_task)), 1 SECONDS)
+		else
+			end_current_task()
+		SStgui.update_uis(src)
+
+/obj/machinery/big_manipulator/proc/validate_all_points()
+	var/list/pickup_to_remove = list()
+	for(var/datum/interaction_point/point in pickup_points)
+		if(!point.is_valid())
+			pickup_to_remove += point
+
+	for(var/datum/interaction_point/point_to_remove in pickup_to_remove)
+		pickup_points.Remove(point_to_remove)
+
+	var/list/dropoff_to_remove = list()
+	for(var/datum/interaction_point/point in dropoff_points)
+		if(!point.is_valid())
+			dropoff_to_remove += point
+
+	for(var/datum/interaction_point/point_to_remove in dropoff_to_remove)
+		dropoff_points.Remove(point_to_remove)
+
+	if(is_operational)
+		update_hud()
+
+/// Attempts to press the power button.
+/obj/machinery/big_manipulator/proc/try_press_on(mob/living/carbon/human/user)
+	if(power_access_wire_cut)
+		balloon_alert(user, "unresponsive!")
+		return
+
+	// Reject activation during stopping task
+	if(current_task == CURRENT_TASK_STOPPING)
+		balloon_alert(user, "stopping in progress!")
+		return
+
+	toggle_power_state(user)
+	if(on)
+		balloon_alert(user, "activated")
+	else
+		balloon_alert(user, "deactivated")
+
+/// Drop the held atom.
+/obj/machinery/big_manipulator/proc/drop_held_atom()
+	if(isnull(held_object))
+		return
+	var/obj/obj_resolve = held_object?.resolve()
+	obj_resolve?.forceMove(get_turf(obj_resolve))
+	finish_manipulation()
+
+/obj/machinery/big_manipulator/ui_interact(mob/user, datum/tgui/ui)
+	if(id_lock)
+		to_chat(user, span_warning("[src] is locked behind ID authentication!"))
+		ui?.close()
+		return
+	if(!anchored)
+		to_chat(user, span_warning("[src] isn't attached to the ground!"))
+		ui?.close()
+		return
```

Comment:
it seems like what you want is just an override of /ui_status()? I suspect the chat messages will not be seen, what's your goal w em?